persistence/sql/persister_consent.go

```diff
@@ -611,6 +608,44 @@ func (p *Persister) VerifyAndInvalidateLogoutRequest(ctx context.Context, verifi
 	})
 }
 
+func (p *Persister) FlushInactiveLoginSessions(ctx context.Context, _ time.Time, limit, _ int) error {
+	return p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
+		// "hydra_oauth2_authentication_request"
+		var lr consent.LoginRequest
+
+		// "hydra_oauth2_consent_request"
+		var cr consent.ConsentRequest
+
+		// "hydra_oauth2_authentication_session"
+		var ls consent.LoginSession
+		query := fmt.Sprintf(`
+			DELETE FROM %[1]s WHERE id in
+			(SELECT id
+			FROM %[1]s
+			WHERE NOT EXISTS
+			    (
+			    SELECT NULL
+			    FROM %[2]s
+			    WHERE %[2]s.login_session_id = %[1]s.id
+			    )
+			AND NOT EXISTS
+			    (
+			    SELECT NULL
+			    FROM %[3]s
+			    WHERE %[3]s.login_session_id = %[1]s.id
+			    )
+			LIMIT %[4]d)
```

Comment:
Please add the expiry to the where clause, otherwise this will remove legitimate sessions.